.github/CONTRIBUTING.md

```diff
@@ -4,72 +4,149 @@ Water.css becomes better for everyone when people like you help make it better!
+
+## Create your pull request
+
+Once you're happy with your changes, you need to **commit** them, create a **changelog** and **submit a pull request**.
+
+A few general rules of thumb about what makes a good pull request:
+
+- Make sure that your pull request covers a small and well defined scope
+- Make small commits with clear and explainful messages
+- Provide a clear description about your contribution on GitHub
+
+When you commit code, Water.css will run some checks to make sure that your changes have no errors and match the project's coding style – a process called [**Linting**](https://www.freecodecamp.org/news/what-is-linting-and-how-can-it-save-you-time). It will also verify that all **colors are accessible**, which means they need to have enough contrast to be easily readable.
+
+If the code you introduced has errors or some of the colors aren't accessible, you'll see an error and the commit will abort.
+You can check whether your changes pass all tests before committing them by running:
 
 ```
-yarn build
+yarn validate
 ```
 
-## How to find issues to work on
+If there are errors, you can try to fix them automatically by running:
 
-If you are new to contributing open-source software, you can starty by picking any relevant issue that is tagged with `good first issue` [there](
-https://github.com/kognise/water.css/contribute).
+```
+yarn lint
+```
 
-Also everyone is welcome to contribute on issues tagged with `help wanted`, you can find it filtered [here](https://github.com/kognise/water.css/issues?q=is%3Aopen+is%3Aissue+label%3A%22help+wanted%22).
+<details>
+  <summary>ℹ A note about puppeteer and WSL</summary>
+  <br>
+  <blockquote>
+    The accessibility checks use puppeteer, a tool that uses Chrome to render websites "headlessly", without a visible interface. In some environments like the <a href="https://aka.ms/wsl">Windows Subsystem for Linux</a>, you'll need to manually configure and run an X-Server in order for puppeteer to work.
+  </blockquote>
+</details>
 
+<br>
 
-## How to make a pull request
+Now that your changes are commited, you'll need to **generate a changelog**:
 
-It is a few general rules of thumb about making pull requests:
+We use a tool called [Changesets](https://github.com/atlassian/changesets) to keep track of what changes have been made between versions. Make sure to create a changelog before creating your pull request!
 
-* Make sure that your pull request covers a small and well defined scope
+Just run this command for each change you made, and then answer a few questions:
 
-* Make small commits with clear and explainful messages
+```
+yarn changeset
+```
 
-* You need to provide a clear description about your contribution on GitHub
+If you're not sure what kind of change you're making, pick "patch".
```

Comment:
We might want to make definitions for a patch, minor, or major change. Our API is pretty small (add one of a few CSS files to your head) so I don't expect us to be making breaking changes often, but maybe our definition of a major change might expand to include some other things like major CSS overhauls. 🤷‍♂️

Probably something to take care of after release.